packages/compiler/server/serverlib.ts

```diff
@@ -1389,58 +1389,84 @@ type SignatureHelpNode =
   | TypeReferenceNode;
 
 function getSignatureHelpNodeAtPosition(
-  file: TypeSpecScriptNode,
+  script: TypeSpecScriptNode,
   position: number
 ): { node: SignatureHelpNode; argumentIndex: number } | undefined {
-  const node = getNodeAtPosition(file, position);
+  // Move back over any trailing whitespace. Otherwise, if there is no
+  // closing paren/angle bracket, we can find ourselves outside the desired
+  // node altogether in cases like `@dec(test, |`.
+  while (position > 0 && isWhiteSpace(script.file.text.charCodeAt(position - 1))) {
```

Comment:
Ideally, this would skip all trailing trivia, not just whitespace. I think this will still fail with something like `@dec(a, b, /* hello * |`.I think the best approach would be to add getCommentAtPosition and use that. This could be a follow-up issue of lower priority. I don't think this use case is anywhere near as important as just a trailing `, ` with no close paren/angle bracket that this fixes.

Reply:
I'm not sure how getCommentAtPosition would help in this case? But agree some kind of previousNonTrivialToken helper would be good, fine if it's future.

Reply:
Filed #2012